### DISCUSSION TITLE
Possibility to have a non-full screen view

### DISCUSSION THREAD

**Discussion by andreaghezzi** [caplin/FlexLayout#336] [Q&A]
Hi everyone!
This is the most powerful layout manager for tabs I've found.
I need to have this FlexLayout in the middle of my page, while above and below it will be other things, such as images and text. Is it possible then to resize the entire Layout component?
An example is here: 
![Schermata da 2022-10-03 12-28-16](https://user-images.githubusercontent.com/26365846/193556375-476fa659-cfbd-45a8-828e-8e3b285073d1.png)

**Comment by sscarduzio** [comment]
This is very important for us as well: we already have a React project with our own sidebar, and would like to render the FlexLayout tabs inside a div, as a regular React component would do.

Please confirm at least if this is even possible?  Or are there any workarounds?

**Comment by nealus** [comment]
yes flexlayout can be put in any div (it will fill the div, which should have position absolute or relative), it handles the container resizing

**Comment by sscarduzio** [comment]
Yes but how? All I get is absolute positions in the generated tabs, and my sidebar and header are covering the tab controls. 


<img width="391" alt="image" src="https://user-images.githubusercontent.com/1327189/197170196-3b88e6d3-d65b-4608-b61e-a5003366e8a4.png">

**Comment by nealus** [comment]
do you have flexlayout inside a container div (which must have position: absolute or relative)?
Internally flexlayout positions tabs etc using absolute positions

**Reply by sscarduzio** [reply]
Yes now I wrapped it inside a relative position div, and I get this (like, no CSS is applied for some reason?)

<img width="629" alt="image" src="https://user-images.githubusercontent.com/1327189/197175385-50cf806c-eafe-4d48-b7e4-ec12445867a8.png">

```js
import React from 'react'
import * as FlexLayout from 'flexlayout-react'

import GraphCard from '../../dashboard/GraphCard'

const json = {
  global: { tabEnableClose: false },
  borders: [
    {
      type: 'border',
      location: 'bottom',
      size: 90,
      children: [
        {
          type: 'tab',
          name: 'four',
          component: 'text',
        },
      ],
    },
    {
      type: 'border',
      location: 'left',
      size: 90,
      children: [],
    },
  ],
  layout: {
    type: 'row',
    weight: 100,
    children: [
      {
        type: 'tabset',
        weight: 50,
        selected: 0,
        children: [
          {
            type: 'tab',
            name: 'One',
            component: 'text',
          },
        ],
      },
      {
        type: 'tabset',
        weight: 50,
        selected: 0,
        children: [
          {
            type: 'tab',
            name: 'Two',
            component: (
              <div style={{ height: '100%' }}>
                <GraphCard></GraphCard>
              </div>
            ),
          },
          {
            type: 'tab',
            name: 'Three',
            component: 'text',
          },
        ],
      },
    ],
  },
}

class App extends React.Component {
  constructor(props) {
    super(props)
    this.state = { model: FlexLayout.Model.fromJson(json) }
  }

  factory(node) {
    const component = node.getComponent()
    if (component === 'text') {
      return (
        <div style={{ height: '100%' }} className="panel">
          Panel {node.getName()}
        </div>
      )
    }
    return node.getComponent()
  }

  render() {
    return <FlexLayout.Layout model={this.state.model} factory={this.factory.bind(this)} />
  }
}
export default App

```

**Reply by sscarduzio** [reply]
I tried adding `import 'flexlayout-react/style/light.css' ` but it's again having the issue with absolute positioning. Maybe we need a relative positioning friendly CSS file?

**Reply by nealus** [reply]
you need a surrounding container div like this:

<div style={{position:"relative"}}>
  <FlexLayout.Layout model={this.state.model} factory={this.factory.bind(this)} />
</div>

**Comment by nealus** [comment]
```
<div style={{position:"relative"}}>
  <FlexLayout.Layout model={this.state.model} factory={this.factory.bind(this)} />
</div>
```

**Reply by sscarduzio** [reply]
I already had that  relative position wrapper in the caller component. Anyway, I reproduced the issue in this sandbox.

https://codesandbox.io/s/zealous-water-yqmbf9?file=/src/TabbedView.js

We have a header, a footer, and a middle div where I'd like the the layout to live.

**Comment by nealus** [comment]
the middle div needs to have a size e.g:

```<div style={{ position: "relative", border:"1px solid blue", height:100 }}>```
so if the flexlayout was not there the container would fill the area you expect

**Reply by sscarduzio** [reply]
now we have this:

<img width="768" alt="image" src="https://user-images.githubusercontent.com/1327189/197216741-b6943660-6339-4f9f-b5e9-e7988977aba6.png">

Still garbled controls, some CSS is missing!

**Comment by nealus** [comment]
add the import 'flexlayout-react/style/light.css';

**Reply by sscarduzio** [reply]
Now it's disappeared, I think it's because that CSS assumes all the positioning is absolute.

<img width="1445" alt="image" src="https://user-images.githubusercontent.com/1327189/197221358-91339acf-f11f-4964-8e63-4991930c73a6.png">

**Reply by sscarduzio** [reply]
are you able to run this sandbox at all?

**Comment by nealus** [comment]
![image](https://user-images.githubusercontent.com/16133809/197223779-685332f5-ec11-4c8a-b557-360eb64e10fa.png)

**Reply by sscarduzio** [reply]
Thank you! It works now 👍 
So the wrapper had to have an explicit height!

**Comment by nealus** [comment]
the containing div must have a size, either explicitly or due to its layout (so it would fill that area even without containing the flexlayout). A flexbox layout (display:flex) might be a good option for your header, center, footer  outer layout seehttps://css-tricks.com/snippets/css/a-guide-to-flexbox/

**Comment by yukosgiti** [comment]
Changing the `.flexlayout__layout` class in the css file to
```css
.flexlayout__layout {
  left: 0;
  top: 0;
  right: 0;
  bottom: 0;
  position: relative;
  width: 100%;
  height: 100%;
  overflow: hidden;
}
```

Seem to work for me. Dragging dropping tabs etc. seems fine. I am just playing with the library though, there might be cases I may have missed.
